src/combinatorics/quiver/path.lean

```diff
@@ -48,6 +60,14 @@ instance {a : V} : inhabited (path a a) := ⟨path.nil⟩
 lemma eq_of_length_zero (p : path a b) (hzero : p.length = 0) : a = b :=
 by { cases p, { refl }, { cases nat.succ_ne_zero _ hzero } }
 
+lemma nil_of_length_zero (p : path a b) (hzero : p.length = 0) :
+  (eq_of_length_zero p hzero).rec_on p = path.nil :=
```

Comment:
Could you use `path.cast` here instead of `eq.rec_on`?

Reply:
As of now, I can't, since the `cast` code is in a file importing `path.lean`, but I think I can actually just get rid of this lemma.

Reply:
As of now I can't, because of the import ordering, but I think getting rid of this lemma isn't a big problem.

Reply:
The `cast.lean` file contains both casts for paths and arrows, but I'm wondering if this should be done differently?

Reply:
Rather than getting rid of the lemma, maybe you could move it into `cast.lean`?